src/plugins/intel_gpu/src/graph/impls/onednn/fully_connected_onednn.cpp

```diff
@@ -140,7 +140,22 @@ struct fully_connected_onednn : typed_primitive_onednn_impl<fully_connected> {
 
         auto input_md = onednn::layout_to_memory_desc(input_layout, dnnl::memory::format_tag::ab, false);
         // TODO: should change format to any. May need a reorder.
-        auto weights_md = onednn::layout_to_memory_desc(weights_layout, dnnl::memory::format_tag::ba);
+        dnnl::memory::desc weights_md;
+        // if weights_layout has data_padding, add strides to memory desc
+        if (weights_layout.data_padding) {
+            dnnl::memory::dims dims;
+            dims.push_back(weights_layout.feature());
+            dims.push_back(weights_layout.get_tensor().count() / weights_layout.feature());
+
+            auto padded_dims = weights_layout.get_padded_dims();
+            dnnl::memory::dims strides;
+            strides.push_back(1);
+            strides.push_back(padded_dims[1]);
+            dnnl::memory::data_type dt = convert_data_type(weights_layout.data_type);
+            weights_md = dnnl::memory::desc(dims, dt, strides);
```

Comment:
* I think it would be better to be placed within `onednn::layout_to_memory_desc`. 
* This function supports `padded_dims[1]` only. Then could you add assert to check `padded_dims[0] == 0`?
* Could you add test case for this fix?

Reply:
Move codes to `onednn::layout_to_memory_desc `
I will update the test case after the oneDNN PR is merged